### PR TITLE
[15.0][IMP] stock_picking_mass_action: allow to configure `check_assign_all` domain

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
It is useful for a easy tweaking of the configuration.

### CODE REVIEW

File: stock_picking_mass_action/models/stock_picking.py [hunk 1]
```diff
@@ -10,9 +10,13 @@ class StockPicking(Model):
     _inherit = "stock.picking"
 
     @api.model
-    def check_assign_all(self):
+    def check_assign_all(self, domain=None):
         """Try to assign confirmed pickings"""
-        domain = [("picking_type_code", "=", "outgoing"), ("state", "=", "confirmed")]
+        if not domain:
```
Comment (rousseldenis) [inline stock_picking_mass_action/models/stock_picking.py:15]: @LoisRForgeFlow Should be the domain overridable or completely new one ?

Comment (LoisRForgeFlow) [reply stock_picking_mass_action/models/stock_picking.py:15]: not sure what is the difference :sweat_smile: : If you set a domain in the cron code you will completely override the domain by a custom one.

Comment (rousseldenis) [reply stock_picking_mass_action/models/stock_picking.py:15]: Yes. I wanted to say, you don't want to keep the 'outgoing' and 'confirmed' filters and extend the domain ?

Comment (LoisRForgeFlow) [reply stock_picking_mass_action/models/stock_picking.py:15]: Yep, the intention is to give full control. And actually, my use case for this module is for a series of internal operations

**Review by rousseldenis — APPROVED** [review]

**Review by JoanSForgeFlow — APPROVED** [review]
LGTM!

**Comment by LoisRForgeFlow** [comment]
Rebased to include fix for test in other module, ready to merge now.

/ocabot merge patch

**Comment by LoisRForgeFlow** [comment]
A similar change was done in v16 (https://github.com/OCA/stock-logistics-workflow/pull/1490) so not really need to fwport.

**Pull request merged**